src/Path.hs

```diff
@@ -110,6 +110,7 @@ data PathParseException
   | InvalidAbsFile FilePath
   | InvalidRelFile FilePath
   | Couldn'tStripPrefixDir FilePath FilePath
+  | NoParent
```

Comment:
Perhaps it can be renamed to `PathException` and we make `PathParseException` into a type alias and deprecate it.